File: code_challenge_solution/c#/87 Higher or Lower, Heads and Tails.cs
```csharp
using System;

namespace HigherOrLowerHeadsOrTails
{
    class Program
    {
        static void Main(string[] args)
        {
            // Guess higher or lower
            Console.WriteLine("Guess if the next number will be higher or lower than the previous one");

            Random random = new Random();
            int previousNumber = random.Next(1, 101);
            Console.WriteLine("First number: " + previousNumber);

            bool guessedCorrectly = true;

            while (guessedCorrectly)
            {
                Console.Write("Will the next number be higher (H) or lower (L)? ");
                string userInput = Console.ReadLine();

                int nextNumber = random.Next(1, 101);
                Console.WriteLine("Next number: " + nextNumber);

                if ((userInput.ToUpper() == "H" && nextNumber > previousNumber) ||
                        (userInput.ToUpper() == "L" && nextNumber < previousNumber))
                {
                    Console.WriteLine("Correct guess! Keep going...\n");
                    previousNumber = nextNumber;
                }
                else
                {
                    Console.WriteLine("Wrong guess! Game over.");
                    guessedCorrectly = false;
                }
            }

            Console.WriteLine("\n\n");

            // Flip coins
            Console.WriteLine("Guess the outcome of coin flips (Heads or Tails)");

            int numberOfFlips = 3;
            int correctGuesses = 0;

            for (int i = 0; i < numberOfFlips; i++)
            {
                Console.Write($"Flip #{i+1}: ");

                int flip = random.Next(2);
                string flipResult = flip == 0 ? "Heads" : "Tails";

                string userInput = Console.ReadLine();

                if (userInput.Equals(flipResult, StringComparison.OrdinalIgnoreCase))
                {
                    Console.WriteLine("Correct guess!");
                    correctGuesses++;
                }
                else
                {
                    Console.WriteLine("Wrong guess!");
                }
            }

            Console.WriteLine($"You guessed {correctGuesses} out of {numberOfFlips} correctly.");
        }
    }
}
```

File: code_challenge_solution/python/87 Higher or Lower, Heads and Tails.py
import random

def higher_or_lower():
    num1 = random.randint(1, 10)
    num2 = random.randint(1, 10)
    
    print("Number 1: ", num1)
    print("Number 2: ", num2)
    
    guess = input("Guess if the next number will be higher or lower (h/l): ")
    
    if guess.lower() == 'h' and num2 > num1:
        print("Correct! The next number is higher.")
    elif guess.lower() == 'l' and num2 < num1:
        print("Correct! The next number is lower.")
    else:
        print("Wrong! The next number is", num2)


def heads_or_tails():
    coin = random.randint(0, 1)
    
    guess = input("Guess heads or tails (h/t): ")
    
    if guess.lower() == 'h' and coin == 0:
        print("Correct! It's heads.")
    elif guess.lower() == 't' and coin == 1:
        print("Correct! It's tails.")
    else:
        print("Wrong! It's", "heads" if coin == 0 else "tails")


higher_or_lower()
heads_or_tails()